util/testutil/testing.go

```diff
@@ -65,3 +65,39 @@ func Equals(tb testing.TB, exp, act interface{}) {
 		tb.FailNow()
 	}
 }
+
+// Nil fails the test if given value is not nil
+func Nil(tb testing.TB, value interface{}) {
+	tb.Logf("%+v %+v", value, value != nil)
+	if !isNil(value) {
+		_, file, line, _ := runtime.Caller(1)
+		fmt.Printf("\033[31m%s:%d: expected nil value, got something %#v\033[39m\n\n", filepath.Base(file), line, value)
+		tb.FailNow()
+	}
+}
+
+// NotNil fails the test if given value is nil
+func NotNil(tb testing.TB, value interface{}) {
+	if isNil(value) {
+		_, file, line, _ := runtime.Caller(1)
+		fmt.Printf("\033[31m%s:%d: expected value, got nothing \033[39m\n\n", filepath.Base(file), line)
+		tb.FailNow()
+	}
+}
+
+func isNil(value interface{}) bool {
+	if value == nil {
+		return true
+	}
+
+	refVal := reflect.ValueOf(value)
+
+	if refVal.Kind() >= reflect.Chan &&
+		refVal.Kind() <= reflect.Slice &&
+		refVal.IsNil() {
+		return true
+	}
+
+	return false
+
+}
```

Comment:
This is already getting pretty magical. What is the motivation for extending the three functions we already have?

To me `Assert(myval == nil)` is just as brief but more explicit and clear than `Nil(myval)` which does magic reflection under the hood.

Reply:
Okay, I did that thinking that it would be easier to test for nil slices too.
Some tests on `discovery/triton/triton_test.go` relies on that behaviour. 
I will remove those helpers.

Reply:
Mh, but a slice that is nil should also evaluate `myNilSlice == nil` to true correctly, no?

Reply:
You're absolutely right, I got tricked.

https://play.golang.org/p/Cuqx00m6Sd
 
Actually it is because I am using an `interface{}` as type in our assertions helpers, that we need to assert of the reflection.

One reason more to remove it.